File: PDF Create Fillable PDF Forms/PHP/Fillable PDF Forms/fillable-pdf-forms.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>PDF Create Fillable PDF Forms - Result</title>
</head>
<body>

<?php 

// Get submitted form data
$apiKey = $_POST["apiKey"]; // The authentication key (API Key). Get your own by registering at https://app.pdf.co/documentation/api

// Prepare URL for HTML to PDF API call
$url = "https://api.pdf.co/v1/pdf/edit/add";

// Prepare requests params
// See documentation: https://apidocs.pdf.co
$parameters = array();

// Direct URL of source PDF file.
$parameters["url"] = "https://bytescout-com.s3-us-west-2.amazonaws.com/files/demo-files/cloud-api/pdf-edit/sample.pdf";

// Name of resulting file
$parameters["name"] = "newDocument";

// If large input document, process in async mode by passing true
$parameters["async"] = false;

// Annotations Strings
$annotations =   '[{
    "text":"sample prefilled text",
    "x": 10,
    "y": 30,
    "size": 12,
    "pages": "0-",
    "type": "TextField",
    "id": "textfield1"
},
{
    "x": 100,
    "y": 150,
    "size": 12,
    "pages": "0-",
    "type": "Checkbox",
    "id": "checkbox2"
},
{
    "x": 100,
    "y": 170,
    "size": 12,
    "pages": "0-",
    "link": "https://bytescout-com.s3-us-west-2.amazonaws.com/files/demo-files/cloud-api/pdf-edit/logo.png",
    "type": "CheckboxChecked",
    "id":"checkbox3"
}]';// JSON string

// Convert JSON string to Array
$annotationsArray = json_decode($annotations, true);

$parameters["annotations"] = $annotationsArray;

// Create Json payload
$data = json_encode($parameters);

// Create request
$curl = curl_init();
curl_setopt($curl, CURLOPT_HTTPHEADER, array("x-api-key: " . $apiKey, "Content-type: application/json"));
curl_setopt($curl, CURLOPT_URL, $url);
curl_setopt($curl, CURLOPT_POST, true);
curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($curl, CURLOPT_POSTFIELDS, $data);

// Execute request
$result = curl_exec($curl);

if (curl_errno($curl) == 0)
{
    $status_code = curl_getinfo($curl, CURLINFO_HTTP_CODE);
    
    if ($status_code == 200)
    {
        $json = json_decode($result, true);
        
        if ($json["error"] == false)
        {
            $resultFileUrl = $json["url"];
            
            // Display link to the file with conversion results
            echo "<div><h2>Result:</h2><a href='" . $resultFileUrl . "' target='_blank'>" . $resultFileUrl . "</a></div>";
        }
        else
        {
            // Display service reported error
            echo "<p>Error: " . $json["message"] . "</p>"; 
        }
    }
    else
    {
        // Display request error
        echo "<p>Status code: " . $status_code . "</p>"; 
        echo "<p>" . $result . "</p>";
    }
}
else
{
    // Display CURL error
    echo "Error: " . curl_error($curl);
}

// Cleanup
curl_close($curl);

?>

</body>
</html>

File: PDF Create Fillable PDF Forms/Python/Fillable PDF Forms/FillablePDFForms.py
import os
import requests # pip install requests

# The authentication key (API Key).
# Get your own by registering at https://app.pdf.co/documentation/api
API_KEY = "**************************************"

# Base URL for PDF.co Web API requests
BASE_URL = "https://api.pdf.co/v1"


def main(args = None):
    fillPDFForm()


def fillPDFForm():
    """Fillable PDF form using PDF.co Web API"""

    # Prepare requests params as JSON
    # See documentation: https://apidocs.pdf.co
    payload = "{\n    \"async\": false,\n    \"encrypt\": true,\n    \"name\": \"newDocument\",\n    \"url\": \"https://bytescout-com.s3-us-west-2.amazonaws.com/files/demo-files/cloud-api/pdf-edit/sample.pdf\",\n    \"annotations\":[        \n       {\n            \"text\":\"sample prefilled text\",\n            \"x\": 10,\n            \"y\": 30,\n            \"size\": 12,\n            \"pages\": \"0-\",\n            \"type\": \"TextField\",\n            \"id\": \"textfield1\"\n        },\n        {\n            \"x\": 100,\n            \"y\": 150,\n            \"size\": 12,\n            \"pages\": \"0-\",\n            \"type\": \"Checkbox\",\n            \"id\": \"checkbox2\"\n        },\n        {\n            \"x\": 100,\n            \"y\": 170,\n            \"size\": 12,\n            \"pages\": \"0-\",\n            \"link\": \"https://bytescout-com.s3-us-west-2.amazonaws.com/files/demo-files/cloud-api/pdf-edit/logo.png\",\n            \"type\": \"CheckboxChecked\",\n            \"id\":\"checkbox3\"\n        }          \n        \n    ],\n    \n    \"images\": [\n        {\n            \"url\": \"bytescout-com.s3-us-west-2.amazonaws.com/files/demo-files/cloud-api/pdf-edit/logo.png\",\n            \"x\": 200,\n            \"y\": 250,\n            \"pages\": \"0\",\n            \"link\": \"www.pdf.co\"\n        }\n        \n    ]\n}"
    # Prepare URL for 'Fillable PDF' API request
    url = "{}/pdf/edit/add".format(BASE_URL)

    # Execute request and get response as JSON
    response = requests.post(url, data=payload, headers={"x-api-key": API_KEY, 'Content-Type': 'application/json'})
    if (response.status_code == 200):
        json = response.json()

        if json["error"] == False:
            #  Get URL of result file
            resultFileUrl = json["url"]
            # Download result file
            r = requests.get(resultFileUrl, stream=True)
            if (r.status_code == 200):
                with open(destinationFile, 'wb') as file:
                    for chunk in r:
                        file.write(chunk)
                print(f"Result file saved as \"{destinationFile}\" file.")
            else:
                print(f"Request error: {response.status_code} {response.reason}")
        else:
            # Show service reported error
            print(json["message"])
    else:
        print(f"Request error: {response.status_code} {response.reason}")

if __name__ == '__main__':
    main()